pkg/vm/vm_core.go

```diff
@@ -268,6 +289,17 @@ func (vm *VirtualMachine) UpdatePc(instruction *Instruction, operands *Operands)
 	return nil
 }
 
+// Updates the values of the RunContext's registers according to the executed instruction
+func (vm *VirtualMachine) UpdateRegisters(instruction *Instruction, operands *Operands) error {
+	if err := vm.UpdateFp(instruction, operands); err != nil {
+		return err
+	}
+	if err := vm.UpdateAp(instruction, operands); err != nil {
+		return err
+	}
+	return vm.UpdatePc(instruction, operands)
+}
+
```

Comment:
We can leave this function where it was before, so that we have `UpdateXX` functions together.